connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -191,6 +192,21 @@ public boolean isSinkConnector() {
         return SinkConnector.class.isAssignableFrom(connector.getClass());
     }
 
+    public ConnectorType getConnectorType() {
+        return isSinkConnector() ? ConnectorType.SINK : ConnectorType.SOURCE;
+    }
+
+    public static ConnectorType getConnectorType(String clsname) {
+        if (null == clsname || clsname.isEmpty())
+            return null;
+        try {
+            return SinkConnector.class.isAssignableFrom(Class.forName(clsname)) ?
+                ConnectorType.SINK : ConnectorType.SOURCE;
+        } catch (Exception e) {
```

Comment:
Might be worth a trace log statement here to capture the exception. However, the bigger issue is that I don't know that this is safe to load the class: the TCCL would have to have been switched to the plugin delegating classloader in order for this to work, and I'm not able to figure out with all the hops.